config/config.go

```diff
@@ -0,0 +1,101 @@
+package config
+
+import (
+	"bytes"
+	_ "embed"
+	"fmt"
+
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+
+	"github.com/onflow/flow-go/config/network"
+)
+
+var (
+	conf = viper.New()
+
+	//go:embed config.yml
+	configFile string
+)
+
+// FlowConfig Flow configuration.
+type FlowConfig struct {
+	NetworkConfig *network.Config `mapstructure:"network-config"`
+}
+
+// Validate validate Flow config.
+func (fc *FlowConfig) Validate() error {
+	err := fc.NetworkConfig.Validate()
+	if err != nil {
+		return fmt.Errorf("failed to validate flow network configuration values: %w", err)
+	}
+	return nil
+}
+
+// DefaultConfig initializes the flow configuration. All default values for the Flow
+// configuration are stored in the config.yml file. These values can be overriden
+// by node operators by setting the corresponding cli flag. DefaultConfig should be called
+// before any pflags are parsed, this will allow the configuration to initialize with defaults
+// from config.yml.
+// Returns:
+//
+//	*FlowConfig: an instance of the network configuration fully initialized to the default values set in the config file
+//	error: if there is any error encountered while initializing the configuration, all errors are considered irrecoverable.
+func DefaultConfig() (*FlowConfig, error) {
+	var flowConf FlowConfig
+	err := Unmarshall(&flowConf)
```

Comment:
```suggestion
	err := conf.Unmarshal(flowConfig, func(decoderConfig *mapstructure.DecoderConfig) {
		// enforce that the default config contains no missing or extraneous fields
		decoderConfig.ErrorUnused = true
		decoderConfig.ErrorUnset = true
	})
```
At least when reading the defaults file (probably not for a user-provided config file, if we choose to support that), checking for missing and extraneous fields would help keep this file neat and tidy.

Reply:
https://github.com/onflow/flow-go/pull/4340/commits/0371afbcf19887f7aacbd83a78cc164c00c23f13

For now we allow unused keys until all flags and configs are moved to the config package.